src/main/java/org/threadly/examples/features/PrioritySchedulerExample.java

```diff
@@ -0,0 +1,146 @@
+package org.threadly.examples.features;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.threadly.concurrent.future.FutureUtils;
+import org.threadly.concurrent.future.ListenableFuture;
+import org.threadly.concurrent.PriorityScheduler;
+import org.threadly.concurrent.TaskPriority;
+
+
+/**
+ * Provides a basic example of the PriorityScheduler feature.
+ * 
+ * @author sreid8 - Sean Reid
+ *
+ */
+public final class PrioritySchedulerExample {
+  
+  /** static reference to the futures created */ 
+  private static List<ListenableFuture<?>> statFutures;
+  
+  /**
+   * Provided a PriorityScheduler that contains some basic ideas
+   * @return -> the PriorityScheduler with tasks added
+   */
+  public static PriorityScheduler getPriorityScheduler() {
+    final int numThreads = Runtime.getRuntime().availableProcessors() * 2;
+    final PriorityScheduler executor = new PriorityScheduler(numThreads);
+    List<ListenableFuture<?>> futures = new ArrayList<ListenableFuture<?>>(numThreads);
+    
+    futures.add(executor.submit(new Runnable() {
+      
+      @Override
+      public void run() {
+        /*
+         * This is the Starvable priority task that will be run.
+         * Starvable tasks run only when no other task needs to run
+         * It can be used to keep track of configuration:
+         * Example use case: closing sockets that no longer have hosts
```

Comment:
This example use case does not make sense for starvable, because if it was actually starved the sockets would end up never closing and you would run out of file handles.

Right now I don't have a an amazing use case for starvable.  The best I have considered is a starvable task could be used as a indication to bring in more work.  Not like pulling from a queue (since usually the task would do that), but maybe it calls out to a micro service to indicate that it is ready for an upstream service to produce more work for it.

I would be okay with ignoring the starvable example if we want for now.